matching-core/src/main/java/com/graphhopper/matching/MapMatching.java

```diff
@@ -113,11 +113,10 @@ public MapMatching(GraphHopper graphHopper, AlgorithmOptions algoOptions) {
         RoutingAlgorithmFactory routingAlgorithmFactory = graphHopper.getAlgorithmFactory(hints);
         if (routingAlgorithmFactory instanceof PrepareContractionHierarchies) {
             ch = true;
-            // I want to use my own instance of FastestWeighting because it uses its own U-Turn-penalty,
-            // but I have to pass a _different_ instance of FastestWeighting to getGraph so it gives me the graph,
-            // (even though this is non-dangerous since it's only about virtual (split) edges (nothing to do with shortcuts),
-            // but it's _still_ a different one than the one which I have to pass to the router itself (see below).
-            routingGraph = graphHopper.getGraphHopperStorage().getCHGraph(((PrepareContractionHierarchies) routingAlgorithmFactory).getWeighting());
+            // I want to use my own instance of FastestWeighting because it uses its own heading penalty,
```

Comment:
like this ? Are there multiple CH graphs ? If not we could just do `getCHGraph()`.

Reply:
Yes, there could be multiple CH graphs like for GH core.